test_gnmi_configdb.py fails and block sanity check

<!--
If you are reporting a new issue, make sure that we do not have any duplicates
already open. You can ensure this by searching the issue list for this
repository. If there is a duplicate, please close your issue and add a comment
to the existing issue instead.

If you suspect your issue is a bug, please edit your issue description to
include the BUG REPORT INFORMATION shown below. If you fail to provide this
information within 7 days, we cannot debug your issue and will close it. We
will, however, reopen it if you later provide the information.

For more information about reporting issues, see
https://github.com/sonic-net/SONiC/wiki#report-issues

---------------------------------------------------
GENERAL SUPPORT INFORMATION
---------------------------------------------------

The GitHub issue tracker is for bug reports and feature requests.
General support can be found at the following locations:

- SONiC Support Forums - https://groups.google.com/forum/#!forum/sonicproject

---------------------------------------------------
BUG REPORT INFORMATION
---------------------------------------------------
Use the commands below to provide key information from your environment:
You do NOT have to include this information if this is a FEATURE REQUEST
-->

**Description**

<!--
Briefly describe the problem you are having in a few paragraphs.
-->
The failed reason is as below:
```
2023-01-23T21:48:43.3970447Z gnmi/test_gnmi_configdb.py::test_gnmi_configdb_incremental_01 
2023-01-23T21:48:43.3977200Z -------------------------------- live log call ---------------------------------
2023-01-23T21:48:43.3979950Z 21:48:43 __init__.pytest_runtest_call             L0040 ERROR  | Traceback (most recent call last):
2023-01-23T21:48:43.3983142Z   File "/usr/local/lib/python2.7/dist-packages/_pytest/python.py", line 1464, in runtest
2023-01-23T21:48:43.3985333Z     self.ihook.pytest_pyfunc_call(pyfuncitem=self)
2023-01-23T21:48:43.3988037Z   File "/usr/local/lib/python2.7/dist-packages/pluggy/hooks.py", line 286, in __call__
2023-01-23T21:48:43.3990294Z     return self._hookexec(self, self.get_hookimpls(), kwargs)
2023-01-23T21:48:43.3993103Z   File "/usr/local/lib/python2.7/dist-packages/pluggy/manager.py", line 93, in _hookexec
2023-01-23T21:48:43.3995348Z     return self._inner_hookexec(hook, methods, kwargs)
2023-01-23T21:48:43.3998695Z   File "/usr/local/lib/python2.7/dist-packages/pluggy/manager.py", line 87, in <lambda>
2023-01-23T21:48:43.4001087Z     firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
2023-01-23T21:48:43.4004049Z   File "/usr/local/lib/python2.7/dist-packages/pluggy/callers.py", line 208, in _multicall
2023-01-23T21:48:43.4006044Z     return outcome.get_result()
2023-01-23T21:48:43.4008469Z   File "/usr/local/lib/python2.7/dist-packages/pluggy/callers.py", line 81, in get_result
2023-01-23T21:48:43.4011422Z     _reraise(*ex)  # noqa
2023-01-23T21:48:43.4014629Z   File "/usr/local/lib/python2.7/dist-packages/pluggy/callers.py", line 187, in _multicall
2023-01-23T21:48:43.4016602Z     res = hook_impl.function(*args)
2023-01-23T21:48:43.4019354Z   File "/usr/local/lib/python2.7/dist-packages/_pytest/python.py", line 174, in pytest_pyfunc_call
2023-01-23T21:48:43.4021717Z     testfunction(**testargs)
2023-01-23T21:48:43.4023924Z   File "/azp/_work/19/s/tests/gnmi/test_gnmi_configdb.py", line 63, in test_gnmi_configdb_incremental_01
2023-01-23T21:48:43.4026152Z     assert interface is not None, "Invalid interface"
2023-01-23T21:48:43.4027828Z AssertionError: Invalid interface
2023-01-23T21:48:43.4028523Z 
```


**Steps to reproduce the issue:**
1. run gnmi/test_gnmi_configdb.py
2.
3.

**Describe the results you received:**


**Describe the results you expected:**


**Additional information you deem important:**
<!--
software/ASIC/Hardware Flatform version and info
-->
    **Output of `show version`:**

    ```
    (paste your output here)
    ```

    **Attach debug file `sudo generate_dump`:**

    ```
    (paste your output here)
    ```

--- a/tests/common/plugins/conditional_mark/tests_mark_conditions.yaml
+++ b/tests/common/plugins/conditional_mark/tests_mark_conditions.yaml
@@ -387,15 +387,6 @@ generic_config_updater/test_incremental_qos.py::test_incremental_qos_config_upda
       - "not any(i in hwsku for i in ['2700', 'Arista-7170-64C', 'montara', 'newport'])"
 
 #######################################
-#####           gnmi              #####
-#######################################
-gnmi/test_gnmi_configdb.py:
-  skip:
-    reason: "We still have issue for config reload command, and we will enable this test after merging the fix."
-    conditions:
-    - https://github.com/sonic-net/sonic-mgmt/issues/7462
-
-#######################################
 #####           http              #####
 #######################################
 http/test_http_copy.py:

--- a/tests/common/plugins/conditional_mark/tests_mark_conditions.yaml
+++ b/tests/common/plugins/conditional_mark/tests_mark_conditions.yaml
@@ -341,7 +341,7 @@ ecmp/test_fgnhg.py:
 #######################################
 everflow/test_everflow_per_interface.py:
   skip:
-    reason: "Skip running on dualtor testbed/unsupported platforms or 
+    reason: "Skip running on dualtor testbed/unsupported platforms or
                   multi-asic due to https://github.com/sonic-net/sonic-buildimage/issues/11776"
     conditions_logical_operator: or
     conditions:

--- a/tests/gnmi/conftest.py
+++ b/tests/gnmi/conftest.py
@@ -3,7 +3,7 @@ import shutil
 
 from tests.common.helpers.assertions import pytest_require as pyrequire
 from tests.common.helpers.dut_utils import check_container_state
-from tests.gnmi.helper import GNMI_CONTAINER_NAME, apply_cert_config, create_ext_conf
+from tests.gnmi.helper import GNMI_CONTAINER_NAME, apply_cert_config, recover_cert_config, create_ext_conf
 from tests.generic_config_updater.gu_utils import create_checkpoint, rollback
 
 SETUP_ENV_CP = "test_setup_checkpoint"
@@ -128,4 +128,4 @@ def setup_gnmi_server(duthosts, rand_one_dut_hostname, localhost):
 
     # Rollback configuration
     rollback(duthost, SETUP_ENV_CP)
-    duthost.shell("systemctl restart %s" % GNMI_CONTAINER_NAME)
+    recover_cert_config(duthost)

--- a/tests/gnmi/helper.py
+++ b/tests/gnmi/helper.py
@@ -40,12 +40,14 @@ def apply_cert_config(duthost):
     duthost.shell(dut_command)
     time.sleep(GNMI_SERVER_START_WAIT_TIME)
 
+
 def recover_cert_config(duthost):
     dut_command = "docker exec %s pkill telemetry" % (GNMI_CONTAINER_NAME)
     duthost.shell(dut_command, module_ignore_errors=True)
     dut_command = "docker exec %s supervisorctl start %s" % (GNMI_CONTAINER_NAME, GNMI_PROGRAM_NAME)
     duthost.shell(dut_command)
     time.sleep(GNMI_SERVER_START_WAIT_TIME)
+
 
 def gnmi_capabilities(duthost, localhost):
     ip = duthost.mgmt_ip

--- a/tests/gnmi/helper.py
+++ b/tests/gnmi/helper.py
@@ -40,6 +40,12 @@ def apply_cert_config(duthost):
     duthost.shell(dut_command)
     time.sleep(GNMI_SERVER_START_WAIT_TIME)
 
+def recover_cert_config(duthost):
+    dut_command = "docker exec %s pkill telemetry" % (GNMI_CONTAINER_NAME)
+    duthost.shell(dut_command, module_ignore_errors=True)
+    dut_command = "docker exec %s supervisorctl start %s" % (GNMI_CONTAINER_NAME, GNMI_PROGRAM_NAME)
+    duthost.shell(dut_command)
+    time.sleep(GNMI_SERVER_START_WAIT_TIME)
 
 def gnmi_capabilities(duthost, localhost):
     ip = duthost.mgmt_ip

--- a/tests/gnmi/test_gnmi_configdb.py
+++ b/tests/gnmi/test_gnmi_configdb.py
@@ -26,7 +26,7 @@ def get_first_interface(duthost):
         return None
     admin_index = status_data[0].split().index('Admin')
     for line in status_data:
-        if "trunk" in line:
+        if "routed" not in line:
             interface_status = line.strip()
             assert len(interface_status) > 0, "Failed to read interface properties"
             sl = interface_status.split()

